src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -131,6 +136,18 @@ export default function IndexPatternFactory(Private, timefilter, Notifier, confi
       }
     };
 
+    // Get the source filtering configuration for that index.
+    // Fields which name appears in the given columns array will not be excluded.
+    self.getSourceFiltering = function (columns) {
+      return {
+        exclude: _(self.getNonScriptedFields())
```

Comment:
I know the `_.chain` is implicit here, but I'm a fan of using it explicitly. Makes it clearer for newcomers.